lib/mpool/src/mclass.c

```diff
@@ -434,28 +434,31 @@ mclass_props_get(struct media_class *const mc, struct mpool_mclass_props *const
     strlcpy(props->mc_path, mc->upath, sizeof(props->mc_path));
 }
 
-merr_t
-mclass_ftw(struct media_class *mc, const char *prefix, struct mpool_file_cb *cb)
+static thread_local struct mpool_file_cb *mclass_ftw_cb;
```

Comment:
I think this should be good for now.. WAL replay is the only user of mclass_ftw() and it calls it in the context of a single thread.